dll/win32/shell32/folders/CRecycleBin.cpp

```diff
@@ -192,10 +192,10 @@ CRecycleBinEnum::~CRecycleBinEnum()
 HRESULT WINAPI CRecycleBinEnum::Initialize(DWORD dwFlags)
 {
     WCHAR szDrive[8];
-    if (GetEnvironmentVariableW(L"SystemDrive", szDrive, _countof(szDrive)))
-        PathAddBackslashW(szDrive);
-    else
-        StringCbCopyW(szDrive, sizeof(szDrive), L"C:\\");
+    if (!GetEnvironmentVariableW(L"SystemDrive", szDrive, _countof(szDrive) - 1))
+        return E_FAIL;
```

Comment:
A DPRINT or FAILED_UNEXPECTEDLY or something like it would be nice, but other than that I agree that failing is the correct thing to do here.

Reply:
`ERR`'s are inserted.